File: TransCoder_tokenized_test_set_functions/java/DECODE_MEDIAN_STRING_ORIGINAL_STRING.java
static String decodeMedianString ( String s ) {
  int l = s . length ( );
  String s1 = "";
  boolean isEven = ( l % 2 == 0 ) ? true : false;
  for ( int i = 0;
  i < l;
  i += 2 ) {
    if ( isEven ) {
      s1 = s . charAt ( i ) + s1;
      s1 += s . charAt ( i + 1 );
    }
    else {
      if ( l - i > 1 ) {
        s1 += s . charAt ( i );
        s1 = s . charAt ( i + 1 ) + s1;
      }
      else {
        s1 += s . charAt ( i );
      }
    }
  }
  return s1;
}

File: results/py/DECODE_MEDIAN_STRING_ORIGINAL_STRING.cpp
string f_filled ( string s ) {
  int l = s . length ( ) ;
  string s1 ;
  if ( ( l % 2 ) == 0 ) {
    bool isEven = true ;
    if ( isEven ) {
      s1 = s [ i ] + s1 ;
      s1 += s [ i + 1 ] ;
    }
    else {
      if ( ( l - i > 1 ) && ( l - i - 1 ) > 1 ) {
        s1 += s [ i ] ;
        s1 = s [ i + 1 ] + s1 ;
      }
      else {
        s1 += s [ i ] ;
      }
    }
  }
  return s1 ;
}
